pymongo/mongo_client.py

```diff
@@ -1700,7 +1706,7 @@ def _process_response(self, reply, session):
         if session is not None:
             session._process_response(reply)
 
-    def server_info(self, session: Optional[client_session.ClientSession] = None) -> Dict[str, Any]:
+    def server_info(self, session: Optional[client_session.ClientSession] = None) -> _DocumentOut:
```

Comment:
This method always returns a dict so isn't the previous return type more correct?

Reply:
Is it not the result of a command call, which is decoded with codec options?

Reply:
It's the result of a command() call _using the default codec_options_ whose document_class is always dict hence this method always returns dict.

Reply:
Okay, I see the distinction now

Reply:
Used a cast to force this to be a dict, since we're limited by https://jira.mongodb.org/browse/PYTHON-3143